plugins/linux-capture/xcomposite-input.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <obs-module.h>
+#include <X11/Xlib.h>
+#include <xcb/xcb.h>
+#include <glad/glad.h>
+#include <glad/glad_glx.h>
+#include <pthread.h>
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#include "xcursor-xcb.h"
+
+struct xcompcap {
+	obs_source_t *source;
+
+	const char *windowName;
+	xcb_window_t win;
+	int crop_top;
+	int crop_left;
+	int crop_right;
+	int crop_bot;
+	bool swapRedBlue;
+	bool include_border;
+	bool exclude_alpha;
+
+	float window_check_time;
+	bool window_changed;
+
+	uint32_t width;
+	uint32_t height;
+	uint32_t border;
+
+	Pixmap pixmap;
+	GLXPixmap glxpixmap;
+	gs_texture_t *tex;
+	gs_texture_t *gltex;
+
+	pthread_mutex_t lock;
+
+	bool show_cursor;
+	bool cursor_outside;
+	xcb_xcursor_t *cursor;
+	// strict_binding determines whether we rebind the GLX Pixmap on every
+	// tick. "Strict binding" is the correct mode of operation, according to
+	// GLX_EXT_texture_from_pixmap. However certain drivers exhibit poor
+	// performance when this is done, so setting this to false allows working
+	// around it.
+	bool strict_binding;
+};
```

Comment:
With the suggestion above, you can move this entire struct to xcomposite-input.c